### ISSUE TITLE
Why convert event/epoch labels from Unicode to bytes?

### ISSUE THREAD

**Issue by jpgill86** [NeuralEnsemble/python-neo#672]
When events and epochs are loaded from proxy objects, as is done by every IO derived from a RawIO (even when `lazy=False`), the dtype of their labels is changed from something like a Unicode string (decided by the RawIO and last handled by `rawio.get_event_timestamps`) to a byte literal. The dtype conversion happens here:

https://github.com/NeuralEnsemble/python-neo/blob/c0107d4524b8b2770644fd26d2fe1ff8d97f6445/neo/io/proxyobjects.py#L414-L415

This causes the labels to be displayed as, for example, `b'My label'` (in Python 3 at least). I'm seeing this show up in ephyviewer's EventList, for example (only when I load the data myself; not with the standalone app, because the data is imported differently there).

What is the rationale for this conversion? Could the conversion be removed so that the unsightly `b''` is not introduced? Or do I need to convert back to Unicode (`labels.astype('U')`) every time I want to display my labels?

I'm currently writing a new RawIO and am perceiving this line of code as an obstacle to controling how my labels are loaded, but maybe I'm missing something! I noticed that the comment above the code seems to be saying it might just be temporary...? Or Unicode labels are planned for a future version of Neo? @samuelgarcia, I think you wrote this; can you provide me with any insights?

**Comment by samuelgarcia** [comment]
Hi Jeffrey.
I don't remember excatly.
I think the rational was retro compatibility, there must be somewhere in unitest a type cheking and I did not to break thing when adding this new rawio layer.

For old version of neo when developed in python in mind label where "str" wich are bytes in fact.
It was a mistake from the begining.
As python will desiapper this year we can go for clear type of labels. But this do not imply only this line.

Because object internally declare there attributes to make automatic unitest type cheking for all IOs.
For instance:
```python
Event._necessary_attrs = [(('times', pq.Quantity, 1), ('labels', np.ndarray, 1, np.dtype('S'))) ...]
```
So labels.astype('S') will be labels.astype('U'), when Event and Epoch will change there internal dtype.

This change could break some piece of code but I think we need to do it.
The object model of neo should refactored soon, next or follinwg version.
This dtype should be change at this moment.
@apdavison @JuliaSprenger @mdenker : I don't remenber if we discuss abotu that ?

**Comment by jpgill86** [comment]
Thanks for the explanation, @samuelgarcia. I will learn to live with the current dtype for now. 😄

**Comment by Kleinjohann** [comment]
@samuelgarcia @jpgill86 I pointed this out earlier and implemented the changes in #647 
It appears I immediately opened a PR without discussing this in an issue first, sorry.
The consensus seemed to be to do this in 0.9

**Comment by jpgill86** [comment]
On the contrary, I should have checked the issues and pull requests before opening this issue! Thanks for your work on this, @Kleinjohann.

Since my question has been answered and #647 is scheduled to solve it eventually, I'm going to close this issue.
